pkg/kube/controllers/extendedjob/errand_reconciler.go

```diff
@@ -99,3 +116,41 @@ func (r *ErrandReconciler) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	return reconcile.Result{}, nil
 }
+
+// injectContainerEnv injects AZ info to container envs
+// errands always have an AZ_INDEX of 1
+func (r *ErrandReconciler) injectContainerEnv(podSpec *corev1.PodSpec) {
+
+	containers := []*corev1.Container{}
+	for i := 0; i < len(podSpec.Containers); i++ {
+		containers = append(containers, &podSpec.Containers[i])
+	}
+	for i := 0; i < len(podSpec.InitContainers); i++ {
+		containers = append(containers, &podSpec.InitContainers[i])
+	}
+	for _, container := range containers {
+		envs := container.Env
+
+		// Default to zone 1, with 1 replica
+		envs = upsertEnvs(envs, EnvCfOperatorAzIndex, "1")
```

Comment:
We can create `envs` outside the `for` block and the `for` block can become:
```go
for _, container := range containers {
  container.Env = envs.DeepCopy()
}
```

This is better due to the upsert being called repeatedly.

Reply:
Not all container envs are the same

Reply:
Oh, I completely missed the fact that we are grabbing the current envs. :expressionless: